src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java

```diff
@@ -53,4 +57,259 @@ public void happyPathVectorSearch() {
+          .then()
+          .statusCode(200)
+          .body("data.documents[0]", jsonEquals(expected))
+          .body("errors", is(nullValue()));
+    }
+
+    @Test
+    public void insertVectorCollectionWithoutVectorData() {
+      String json =
+          """
+        {
+           "insertOne": {
+              "document": {
+                  "_id": "10",
+                  "name": "Coded Cleats",
+                  "description": "ChatGPT integrated sneakers that talk to you"
+              }
+           }
+        }
+        """;
+
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("status.insertedIds[0]", is("10"))
+          .body("data", is(nullValue()))
+          .body("errors", is(nullValue()));
+
+      json =
+          """
+        {
+          "find": {
+            "filter" : {"_id" : "10"}
+          }
+        }
+        """;
+      String expected =
+          """
+        {
+          "_id": "10",
+          "name": "Coded Cleats",
+          "description": "ChatGPT integrated sneakers that talk to you"
+        }
+        """;
+
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.documents[0]", jsonEquals(expected))
+          .body("errors", is(nullValue()));
+    }
+
+    @Test
+    public void insertEmptyVectorData() {
+      String json =
+          """
+        {
+           "insertOne": {
+              "document": {
+                  "_id": "Invalid",
+                  "name": "Coded Cleats",
+                  "description": "ChatGPT integrated sneakers that talk to you",
+                  "$vector": []
+              }
+           }
+        }
+        """;
+
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("data.document", is(nullValue()))
+          .body("status", is(nullValue()))
+          .body("errors", is(notNullValue()))
+          .body("errors", hasSize(1))
+          .body("errors[0].message", startsWith("$vector field can't be empty"))
+          .body("errors[0].errorCode", is("SHRED_BAD_VECTOR_SIZE"))
+          .body("errors[0].exceptionClass", is("JsonApiException"));
+    }
+
+    @Test
+    public void insertInvalidVectorData() {
```

Comment:
I think you were already planning to add such a test, but case of inserting array with too many (or too few) floats seems to be missing.

Reply:
This need changes in the base image, it's not sync'd for the size validation. Will do different PR for this case.